Feature/homepage

realzzkevin opened the pull request.

403392442 committed:
Branch created, component initial file created.

NobodyLand committed:
Implement user schema

403392442 committed:
test file initialed

NobodyLand committed:
create the api file

Commit:
first commit

PHANTOMS0308 committed:
Test out the Pull Request

NobodyLand committed:
implememt create user in database but database not working well

NobodyLand committed:
Implement database store

Commit:
add unit test

Commit:
add component header footer test

NobodyLand committed:
Check if email existed before creating new user

NobodyLand committed:
Create test files

NobodyLand committed:
install node-mocks-http in order to mock api request and response in test

403392442 committed:
need to implement test and new post content

zhiyuan-Z committed:
add unit tests for sign up form

zhiyuan-Z committed:
add sign up form component and sign up page

Commit:
rewrite on MUI

403392442 committed:
need to implement test and new post content

Commit:
update based on CR

PHANTOMS0308 committed:
Merge pull request #3 from jacklu98/feature/entry-page

Ticket 5: Entry Page

zhiyuan-Z committed:
add mui theme provider and add default font

zhiyuan-Z committed:
Merge branch 'ticket-24/add-mui-theme' into dev

realzzkevin committed:
change to layout

realzzkevin committed:
Merge branch 'main' of https://github.com/realzzkevin/Project-clone into feature/homepage

NobodyLand committed:
implement validation function test

NobodyLand committed:
Merge https://github.com/VijayIyer/skool-clone into user-api

Commit:
merge from ticket

403392442 committed:
need to implement test and new post content

realzzkevin committed:
use menu for dropdown

403392442 committed:
need to implement test and new post content

KanyaLee committed:
api login endpoint + test

403392442 committed:
test code finished, need to merge the editor code

realzzkevin committed:
new components

NobodyLand committed:
Unable to implement test for next js api

403392442 committed:
test code finished, need to merge the editor code

realzzkevin committed:
tabbar added

zhiyuan-Z committed:
add additional validation and corresponding unit tests

zhiyuan-Z committed:
Merge branch 'dev' into ticket-1/implement-sign-up-form-frontend

PHANTOMS0308 committed:
Finish the Text Editor Component(beta)

PHANTOMS0308 committed:
Merge branch 'dev' into feature/text-editor

PHANTOMS0308 committed:
Feature: Text Editor

1. Text Editor will be used in Post and Comment UI
2. Test coverage is not high, due to inconsistent Selection and Range in Jest and RTL
3. The component is functionally working, but requires further refactor

403392442 committed:
test code finished, need to merge the editor code

403392442 committed:
need to add comments

xusde committed:
basic functionalities finished

403392442 committed:
fixed some bug

Commit:
add reply interactions for comment section

NobodyLand committed:
Implement test case for sign up api

realzzkevin committed:
styling...

realzzkevin committed:
button styling added

realzzkevin committed:
create PR

PHANTOMS0308 committed:
Hot Fix: NextJS API Testing Error

- The cause of API testing error was the mal-configuration of test environment of Jest.
- The current solution is to separate testing of client side(Pages and components, including server rendered ones) and server side(API)
- To run server tests, you type npm run test:server. Similarly, for client test, it is npm run test:server.
- Also you need to declare test type when naming your test, ending with client.test.tsx or server.test.ts
- Try to incorporate in your ticket to see how this work.

realzzkevin committed:
create alter layout

PHANTOMS0308 commented on components/Layout/searchbar/index.tsx:
```diff
@@ -0,0 +1,36 @@
+  const [searchToken, setSearchToken] = useState<string | null>(null);
+  function handleChange() {}
+  return (
+    <FormControl className={`${style.navbar_search}`}>
```
Make sure you understand the difference and use case of `FormControl` and `TextField`. Usually when you use TextField, it automatically include `FormControl` and other low-level MUI components.

PHANTOMS0308 commented on components/Layout/searchbar/index.tsx:
```diff
@@ -0,0 +1,36 @@
+      <TextField
+        variant="outlined"
+        placeholder="Search..."
+        inputProps={{
```
`InputProps` not `inputProps`

PHANTOMS0308 commented on components/Layout/searchbar/index.tsx:
```diff
@@ -0,0 +1,36 @@
+            </InputAdornment>
+          ),
+          endAdornment: (
+            <InputAdornment position="end">
```
If you have time, maybe here is where you can use the `IconButton` you import, and when on click will reset the input state to empty string

PHANTOMS0308 reviewed (commented):
Well done, before you proceed, there are a few things to consider

- If you have components that defined specifically for another component, it is better to have a `/components` folder inside that bigger component. For example you may have `/Layout/components` so the folder structure is more clear.
- Make sure **all** component folder name to be capitalized to maintain the naming consistency
- Since we using module css, then you could have css in each component folder. In your folders I see many components are taking styles from the parent component `NavBar`. Unless parent and child has relation that needed to be reflected in the CSS, it is better practice to localize the styling file for better readability.
- Another thing to consider is that maybe you don't want to have different layout for different page. If you look at the Skool website, they even have very inconsistent behavior in the header. For example, if you go from Home Page directly to Setting Page, you will not see the search bar, then going to the Profile Page from the Setting Page, the header remain same, no search bar. However if you going from Home Page directly to Profile, you actually see the search bar. This inconsistency is pretty weird, and annoying. Also after thinking about Skool's design, once you are in profile or Setting, you lose the visuals of within-community tab navigation, which I found very confused and maybe that is not something we should replicate. Let me know how you think about it.

realzzkevin committed:
fixed search bar

realzzkevin committed:
create sub module.css file

PHANTOMS0308 committed:
Update mongoClient.ts

Add a clear Error message to the `mongoClient.ts` when `MONGODB_URI` failed to load into the `process.env`

chenyou-H committed:
Merge pull request #18 from PHANTOMS0308/hotfix/api-testing-error

Hotfix: Enhancements in Testing and MongoDB Connection

NobodyLand committed:
Merge https://github.com/VijayIyer/skool-clone into user-api

NobodyLand committed:
modify CRUD mehtod name to make it more self-splanatory

NobodyLand committed:
Change user schema key names fo follow camelCase

PHANTOMS0308 committed:
Merge branch 'dev' into new-post-input-Yuanda

NobodyLand committed:
Change some variable names to camelCase
remove json.parse in the api, because next js router can handle it when clinet set their content type to json

PHANTOMS0308 committed:
Merge pull request #13 from 403392442/new-post-input-Yuanda

Ticket 9 - New Post Input

This ticket will be refactored, thus no code review required

NobodyLand committed:
comment out login helper functions

NobodyLand committed:
rm login helper functions bc login api service is completed by other developer

NobodyLand committed:
should handle error for client when client is unable to connect with databse

NobodyLand committed:
Add more rules for validation in order to match frontend validation

NobodyLand committed:
Place all utl functions in a folder

NobodyLand committed:
modiy my test case based on new test rules, environment and so on

NobodyLand committed:
Only return object id when successfully creare an object

NobodyLand committed:
caplize user names first letter

NobodyLand committed:
centralized response logic

NobodyLand committed:
Add a test formatter test

NobodyLand committed:
Rm type test

NobodyLand committed:
Rm some comments

NobodyLand committed:
Some mege conflict due to prettier

NobodyLand committed:
Change prettier setting to avoid future merge conflict

NobodyLand committed:
Merge remote-tracking branch 'upstream/dev' into user-api

NobodyLand committed:
Install the most update package

PHANTOMS0308 committed:
Merge branch 'dev' into ticket-1/implement-sign-up-form-frontend

PHANTOMS0308 committed:
Merge pull request #8 from zhiyuan-Z/ticket-1/implement-sign-up-form-frontend

Ticket 1/implement sign up form frontend

Commit:
initial commit

PHANTOMS0308 committed:
Hotfix: Merge Conflict from Pull Request`

realzzkevin committed:
modified search component

realzzkevin committed:
captalized component name

PHANTOMS0308 committed:
Delete Dark Mode in global.css

PHANTOMS0308 committed:
Merge pull request #20 from PHANTOMS0308/hotfix/merge-conflict

Hotfix: Merge Conflict from Pull Request`

NobodyLand committed:
Merge remote-tracking branch 'upstream/dev' into user-api

PHANTOMS0308 committed:
Merge pull request #15 from chenyou-H/user-api

Ticket 2: Develop backend API endpoints for user registration and storing user data

realzzkevin committed:
Merge branch 'dev' of https://github.com/VijayIyer/skool-clone into feature/homepage

zhiyuan-Z committed:
Merge branch 'dev' of https://github.com/VijayIyer/skool-clone into login-form

zhiyuan-Z committed:
add css reset

zhiyuan-Z committed:
separate login form to a stand-alone component

Commit:
add Group model and test

zhiyuan-Z committed:
Merge pull request #16 from xusde/login

Finish login form component

zhiyuan-Z committed:
Merge upstream dev branch into lynnda's branch

zhiyuan-Z committed:
refactor unit test for login to use Mongoose, define util functions in lib, modify test cases

zhiyuan-Z committed:
refactor api responses and allow login API response to store jwt token in cookies

zhiyuan-Z committed:
add sample JWT secret to example environment variables

chenyou-H committed:
Merge pull request #6 from KanyaLee/lynnda

Ticket 4: Login API

Commit:
models for group

Commit:
update based on env

Commit:
pull latest

Commit:
Merge remote-tracking branch 'upstream/dev' into feature/group-model-api

Commit:
delete for conflict

Commit:
resolve conflict

Commit:
refine naming in the group model

PHANTOMS0308 committed:
Merge Feature/Group-Model

A tested Group Model to reference for.

realzzkevin committed:
debug

realzzkevin committed:
styling change

realzzkevin committed:
styling update

realzzkevin committed:
merging with dev
